laurat007/assignment3/Word.java

```diff
@@ -0,0 +1,113 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ *
+ * @author Laura
+ * 
+ *  This program returns the words that can be formed using the letters in a 
+ * given grid by moving left, right, up, down and on diagonals.
+ *  Moving in the grid is similar to a backtracking from each Cell which stops
+ * if the current word formed is not a prefix, and if it finds a word it is added
+ * to solution.
+ * 
+ */
+
+public class Word {
+    
+    static TreeSet<String> words(Grid G) {
+        TreeSet<String> solution = new TreeSet<>();
+        boolean[][] visited = new boolean[G.rows][G.columns];
+        
+        // start search from each existing cell
+        for(int i = 0; i < G.rows; i++) {
+            for(int j = 0; j < G.columns; j++) {
+                backtrack(i, j, G, solution, 
+                    Character.toString(G.grid[i][j]), visited);
+                visited[i][j] = false;
```

Comment:
I think it would be better to manipulate visited[i][j] in just one consistent place. Right now you are setting it to true in the beginning of backtrack but setting it back to false outside of it. Decide which method should perform both operations, that would make it much easier to follow.